libgalois/include/katana/PropertyGraph.h

```diff
@@ -767,6 +819,620 @@ CreateSymmetricGraph(PropertyGraph* pg);
 KATANA_EXPORT Result<std::unique_ptr<PropertyGraph>>
 CreateTransposeGraphTopology(const GraphTopology& topology);
 
+// TODO(amber): add NodeLabelVecT
+// template <typename NodeLabelVecT, typename EdgeLabelVecT>
```

Comment:
Will this be part of the public API? We are trying to keep templates out of the public untyped API. It makes Python integration much easier and provides a nice clean separation between organization of data and the specific data interpretation.

Reply:
No. This is leftovers from earlier WIP. Sorry about that.
That said, I might introduce templates in such a way that  they don't make their way to Python. The issue at hand is that I don't want dynamic polymorphism/virtual methods for Graph topology since these are short frequent function calls. I might use static polymorphism via templates. In the end, you will get family of topologies and correspondingly a family of PropertyGraph "views". I'll update the RFC and we can discuss this further there. (Thanks for the comments on RFC, I'll respond to those as well).

Reply:
I don't fully understand what you mean here. But I see your general point. One option might be to use static polymorphism in C++, but still have the virtual functions so we don't need to create a bunch of different cases and potentially have combinatorial explosion in the Python wrappers. I did some testing and if you mark the subclass methods as final then they can be inlined just like a non-virtual function if you have a reference to that time directly. See: https://github.com/KatanaGraph/katana-enterprise/blob/69e3de6fb7c6878e8e9024b2aff6956aee96b25d/libdistgalois/include/katana/WorkDetection.h